### ISSUE TITLE
Rework default interface for Problem class

### ISSUE THREAD

**Issue by tlunet** [Parallel-in-Time/pySDC#254]
Following discussion #253, set a more generic interface for the problem class for more generic ODE, with a default implementation considering the simple generic ODE system

```math
\frac{dU}{dt} = f(U,t)
```

Add full documentation in the Problem class.

:warning: this should probably be done once #244 is merged into `v6`, and before going through the associated rework of all problems implementations.
